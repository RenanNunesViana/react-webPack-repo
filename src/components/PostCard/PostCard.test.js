import { render, screen, describe, it, expect } from '@testing-library/react';
import { PostCard } from '.';
import React from 'react'

const props = {
  title: 'just a test',
  cover: 'just a fake url',
  body: 'just a litle test',
  id: 1,
};

describe('< PostCard />', () => {
  it('should have a title', () => {
    render(<PostCard {...props} />);

    const postCard = screen.getByRole('heading', { name: /just a test/i });

    expect(postCard).toBeInTheDocument();
  });

  it('should have a cover', () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole('img', { name: /just a test/i })).toBeInTheDocument();
  });

  it('should have a body', () => {
    render(<PostCard {...props} />);

    expect(screen.getByText('just a litle test')).toBeInTheDocument();
  });

  it('should match a snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
