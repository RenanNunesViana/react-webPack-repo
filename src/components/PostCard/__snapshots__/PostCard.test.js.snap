// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< PostCard /> should match a snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="just a test"
    src="just a fake url"
  />
  <div
    class="post-content"
  >
    <h2>
      just a test
       
      1
    </h2>
    <p>
      just a litle test
    </p>
  </div>
</div>
`;
